### PR TITLE
XMLParser-File

### PR DESCRIPTION
**Pull request by jgottfriedMSC**
- Implemented generic Dict to XML Function
- Implemented convertISSDBKeyToXML function

# TODO
- convertAstrosOnISSToXML -> Lawand
- convertRSSFeedToXML -> Justin
- convertGeoJsonToXML -> Justin & Lawand

### CODE REVIEW

File: Backend/Core/XMLParser.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    timeValueElem = Element("timeValue")
+
+    for key in requestData:
+
+        if 'time' in timeValueElem.attrib:
+            if timeValueElem.attrib["time"] == key.timeValue:
+                if key.key == 'longitude':
+                    keyElem = Element(key.key)
+                    keyElem.text = str(struct.unpack('f', key.value)[0])
```
Comment (lkno0705) [inline Backend/Core/XMLParser.py:66]: Why do you need Struct.unpack()?

File: Backend/Core/XMLParser.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+def reformatData(requestData, requestName):
+    functions = {
+        'ISSpos': convertISSPosToXML,
+        'ISSDB': convertISSDBKeyToXML
+        # List of Requests
+    }
+
+    return functions.get(requestName)(requestData)
+
+print(tostring(reformatData(l, "ISSDB")))
```
Comment (lkno0705) [inline Backend/Core/XMLParser.py:112]: Please comment out unnecessary print-statements

File: Backend/Core/XMLParser.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    elem.append(requestChild)
+
+    dataChild = Element("data")
+
+    timeValueElem = Element("timeValue")
+
+    for key in requestData:
+
+        if 'time' in timeValueElem.attrib:
+            if timeValueElem.attrib["time"] == key.timeValue:
```
Comment (lkno0705) [inline Backend/Core/XMLParser.py:63]: I think you could save some calculation steps here. As I said, the DBKeys()-objects are in an order, which means that you can look at them in pairs of 2. The only thing which isn't ordered is the key (longitude, latitude). So it should work if you iterate over the keys with a step of 2. You also shouldn't need so many if-clauses.

File: Backend/Core/XMLParser.py (hunk 1 above)
Comment (lkno0705) [reply Backend/Core/XMLParser.py:66]: Also the return values are incorrect. b'1234' != 1.6688933612840628e-07. Also you don't need to bother with byte-Strings anymore. Now the DB returns normal strings, not byte-Strings.

File: Backend/Core/XMLParser.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                    keyElem = Element(key.key)
+                    keyElem.text = str(struct.unpack('f', key.value)[0])
+                    timeValueElem.append(keyElem)
+                    if len(timeValueElem) == 2:
+                        dataChild.append(timeValueElem)
+                        timeValueElem = Element("timeValue")
+
+        else:
+            timeValueElem.attrib = {"time": key.timeValue}
+            if key.key == 'longitude':
```
Comment (lkno0705) [inline Backend/Core/XMLParser.py:81]: I don't think you need this if-clause. The code which gets executed is the same as in the else-clause.

File: Backend/Core/XMLParser.py (hunk 1 above)
Comment (jgottfriedMSC) [reply Backend/Core/XMLParser.py:66]: Resolved!

File: Backend/Core/XMLParser.py (hunk 2 above)
Comment (jgottfriedMSC) [reply Backend/Core/XMLParser.py:112]: Resolved!

File: Backend/Core/XMLParser.py (hunk 3 above)
Comment (jgottfriedMSC) [reply Backend/Core/XMLParser.py:63]: Resolved!

File: Backend/Core/XMLParser.py (hunk 4 above)
Comment (jgottfriedMSC) [reply Backend/Core/XMLParser.py:81]: Resolved!

**Pull request merged**